modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -2778,6 +2823,327 @@ SUB_SIMD(float, float)
 
 #undef SUB_SIMD
 
+//-------------------------
+//
+// Fluid kernels: ConvertTo
+//
+//-------------------------
+
+CV_ALWAYS_INLINE void store_i16(ushort* outx, const v_uint16& res)
+{
+    vx_store(outx, res);
+}
+
+CV_ALWAYS_INLINE void store_i16(short* outx, const v_uint16& res)
+{
+    vx_store(outx, v_reinterpret_as_s16(res));
+}
+
+CV_ALWAYS_INLINE void store_i16(ushort* outx, const v_int16& res)
+{
+    vx_store(outx, v_reinterpret_as_u16(res));
+}
+
+CV_ALWAYS_INLINE void store_i16(short* outx, const v_int16& res)
+{
+    vx_store(outx, res);
+}
+
+CV_ALWAYS_INLINE void convertto_simd_nocoeff_impl(const float* inx, uchar* outx)
+{
+    constexpr int nlanes = v_uint8::nlanes;
+
+    v_int32 a1 = v_round(vx_load(inx));
+    v_int32 a2 = v_round(vx_load(&inx[nlanes/4]));
+    v_int32 a3 = v_round(vx_load(&inx[nlanes/2]));
+    v_int32 a4 = v_round(vx_load(&inx[3*nlanes/4]));
+
+    v_int16 r1 = v_pack(a1, a2);
+    v_int16 r2 = v_pack(a3, a4);
+
+    vx_store(outx, v_pack_u(r1, r2));
+}
+
+template<typename SRC>
+CV_ALWAYS_INLINE
+typename std::enable_if<std::is_same<SRC, short>::value ||
+                        std::is_same<SRC, ushort>::value, void>::type
```

Comment:
it takes a lot of time on review to expand this conditions in mind. Also i see that  these SFINAE repeat again and again

I suggest to introduce MACRO with such condition incapsulation in the way, eventually:
```
typename std::enable_if<SHORT_OR_USHORT>::type 
convertto_simd_nocoeff_impl(...)
<and other functions with similar SFINAE> 
...
#define SHORT_OR_USHORT std::is_same<SRC, short>::value ||  std::is_same<SRC, ushort>::value
```
etc

Reply:
Applied.